src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -325,7 +325,21 @@ internal bool IsRuntimeAsyncEnabledIn(MethodSymbol method)
                 return false;
             }
 
-            return method switch
+            if (symbol is not MethodSymbol method)
+            {
+                return false;
+            }
+
+            var methodReturn = method.ReturnType.OriginalDefinition;
+            if (!ReferenceEquals(methodReturn, GetSpecialType(InternalSpecialType.System_Threading_Tasks_Task))
+                && !ReferenceEquals(methodReturn, GetSpecialType(InternalSpecialType.System_Threading_Tasks_Task_T))
+                && !ReferenceEquals(methodReturn, GetSpecialType(InternalSpecialType.System_Threading_Tasks_ValueTask))
+                && !ReferenceEquals(methodReturn, GetSpecialType(InternalSpecialType.System_Threading_Tasks_ValueTask_T)))
+            {
+                return false;
```

Comment:
Is there a corresponding update to the design doc? Or should we have a follow-up comment to make the `void`-returning method scenario work at some point?

Reply:
Need to update the design doc, the runtime-side is very clear that only Task/ValueTask methods can be runtime async.